.github/workflows/main.yml

```diff
@@ -0,0 +1,63 @@
+name: main
+
+on: [push, pull_request]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.11']
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - uses: Gr1N/setup-poetry@v8
+
+    - name: Check dependencies
+      run: make doctor
+
+    - uses: actions/cache@v4
+      with:
+        path: .venv
+        key: ${{ runner.os }}-poetry-${{ hashFiles('poetry.lock') }}
+
+    - name: Install dependencies
+      run: make install
+
+    - name: Check everything
+      run: make all
+
+    - name: Checkout demo repo
+      if: github.ref_name == 'main' && github.event_name != 'pull_request'
+      uses: actions/checkout@v4
+      with:
+        repository: jacebrowning/template-python-demo
+        path: temp
+
+    - name: Test template in demo repo
+      if: github.ref_name == 'main' && github.event_name != 'pull_request'
+      run: |
+        # Configure Git with GHA information, for more info see:
+        # https://github.com/actions/checkout?tab=readme-ov-file#push-a-commit-using-the-built-in-token
+        git config user.name "github-actions[bot]"
+        git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
```

Comment:
Where does this value come from?

Reply:
This is the unique user ID of the official GHA bot account, taken from the link I put as a comment on L48: https://github.com/actions/checkout?tab=readme-ov-file#push-a-commit-using-the-built-in-token

That doc in turn references the user ID directly from the GH API https://api.github.com/users/github-actions%5Bbot%5D.

Here's another discussion which may explain better: https://github.com/orgs/community/discussions/26560#discussioncomment-3252340